src/modules/sensors/rc_update.cpp

```diff
@@ -350,9 +350,8 @@ RCUpdate::rc_poll(const ParameterHandles &parameter_handles)
 			manual.z = math::constrain(_filter_throttle.apply(manual.z), 0.f, 1.f);
 
 			if (_parameters.rc_map_flightmode > 0) {
-
 				/* the number of valid slots equals the index of the max marker minus one */
-				const int num_slots = manual_control_setpoint_s::MODE_SLOT_MAX;
+				const int num_slots = manual_control_setpoint_s::MODE_SLOT_6;
```

Comment:
Can you use:
```suggestion
				const int num_slots = manual_control_setpoint_s::MODE_SLOT_ARRAY_SIZE - 1;
```

Reply:
makes sense!

Reply:
Did it in the force-push: https://github.com/PX4/Firmware/compare/1e0e537396d3f04e5fae7e4c9997c8e8702e2363..8a8d35b474e1fd9a0a8495726097a422457f7a17